src/hooks/use-photo-filter.tsx

```diff
@@ -0,0 +1,19 @@
+import { useQueryClient } from '@tanstack/react-query';
+import { Image } from 'types/unsplash-types';
+
+type UsePhotoFilterr = () => (name: string, value: string) => void;
+type ChangeFilter = (name: string, value: string) => void;
+
+export const usePhotoFilter: UsePhotoFilterr = () => {
+  const queryClient = useQueryClient();
+  const changeFilter: ChangeFilter = (name, value) => {
+    localStorage.setItem(name, value);
+    queryClient.setQueryData(['photos'], (data: { pageParams: number[]; pages: Image[][] }) => ({
+      pages: data.pages.length ? [data.pages[0]] : [],
+      pageParams: data.pageParams.length ? [data.pageParams[0]] : [],
+    }));
+    queryClient.invalidateQueries({ queryKey: ['photos'] });
+  };
+
+  return (name, value) => changeFilter(name, value);
```

Comment:
1. I don't see here useQuery with `['photo-filter]` key
2. u should return data from useQuery where u will store filter data and update function

`const {data: filter} = useQuery({queryKey: ['photo-filter], staleTime: Infinity}`

Reply:
rewrited